internal/install/recipe_installer.go

```diff
@@ -316,31 +364,142 @@ func (i *RecipeInstaller) failMessage(componentName string) error {
+
+			recipes = append(recipes, *recipe)
+		}
+	} else if i.RecipeNamesProvided() {
+		for _, n := range i.RecipeNames {
+			log.Debugln(fmt.Sprintf("Attempting to match recipeName %s.", n))
+			r := findRecipeInRecipes(n, recipesForPlatform)
+			if r != nil {
+				// Ignore anything that does not match the requested name.
+				if r.Name == n {
+					log.Debugln(fmt.Sprintf("Found recipe from name %s.", n))
+					recipes = append(recipes, *r)
+				} else {
+					log.Debugln(fmt.Sprintf("Ignoring recipe, name %s does not match.", r.Name))
+				}
+			}
+		}
+	}
+
+	return recipes, nil
+}
+
+func (i *RecipeInstaller) recipeFromPath(recipePath string) (*types.OpenInstallationRecipe, error) {
+	recipeURL, parseErr := url.Parse(recipePath)
+	if parseErr == nil && recipeURL.Scheme != "" {
+		f, err := i.recipeFileFetcher.FetchRecipeFile(recipeURL)
+		if err != nil {
+			return nil, fmt.Errorf("could not fetch file %s: %s", recipePath, err)
+		}
+		return f, nil
+	}
+
+	f, err := i.recipeFileFetcher.LoadRecipeFile(recipePath)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("could not load file %s: %s", recipePath, err)
+	}
+
+	return f, nil
+}
+
+func (i *RecipeInstaller) promptUserSelect(recipes []types.OpenInstallationRecipe) ([]types.OpenInstallationRecipe, []types.OpenInstallationRecipe, error) {
+	if len(recipes) == 0 {
+		return []types.OpenInstallationRecipe{}, []types.OpenInstallationRecipe{}, nil
 	}
 
-	i.status.RecipeAvailable(*r)
+	names := []string{}
+	for _, r := range recipes {
+		names = append(names, r.DisplayName)
+	}
+
+	selected, promptErr := i.prompter.MultiSelect("Please choose from the following instrumentation to be installed:", names)
+	if promptErr != nil {
+		return nil, nil, promptErr
+	}
+
+	fmt.Println()
+
+	var selectedRecipes, unselectedRecipes []types.OpenInstallationRecipe
+	for _, r := range recipes {
+		if utils.StringInSlice(r.DisplayName, selected) {
+			selectedRecipes = append(selectedRecipes, r)
+		} else {
+			unselectedRecipes = append(unselectedRecipes, r)
+		}
+	}
 
-	return r, nil
+	return selectedRecipes, unselectedRecipes, nil
 }
 
-func (i *RecipeInstaller) fetch(ctx context.Context, m *types.DiscoveryManifest, recipeName string) (*types.OpenInstallationRecipe, error) {
-	r, err := i.recipeFetcher.FetchRecipe(ctx, m, recipeName)
-	if err != nil {
-		log.Errorf("error retrieving recipe %s: %s", recipeName, err)
-		return nil, err
+func findRecipeInRecipes(name string, recipes []types.OpenInstallationRecipe) *types.OpenInstallationRecipe {
+	for _, r := range recipes {
+		if r.Name == name {
+			return &r
+		}
 	}
 
-	if r == nil {
-		return nil, fmt.Errorf("recipe %s not found", recipeName)
+	return nil
+}
+
+func orderAndDedupeRecipes(recipes []types.OpenInstallationRecipe) []types.OpenInstallationRecipe {
+	deduped := dedupeRecipes(recipes)
+	ordered := orderRecipes(deduped)
+
+	return ordered
+}
+
+func dedupeRecipes(recipes []types.OpenInstallationRecipe) []types.OpenInstallationRecipe {
```

Comment:
The dedup logic is complex. 
We need to find the best match looking at the recipe installTarget and discoveryManifest. 
In case of multiple findings, the most matching one should be returned. Any candidate not matching any of the recipe criteria should be discarded.